src/services/item/plugins/app/appAction/service.ts

```diff
@@ -1,9 +1,10 @@
-import { AppAction, PermissionLevel } from '@graasp/sdk';
+import { PermissionLevel } from '@graasp/sdk';
 
 import HookManager from '../../../../../utils/hook';
 import { Repositories } from '../../../../../utils/repositories';
 import { validatePermission } from '../../../../authorization';
 import { ManyItemsGetFilter, SingleItemGetFilter } from '../interfaces/request';
+import { AppAction } from './appAction';
```

Comment:
Not sure about this, shouldn't the service layer deal with the business-level types from SDK?

For instance in the lines 12-17 below:
```ts
hooks = new HookManager<{
    post: {
      pre: { appAction: Partial<InputAppAction>; itemId: string };
      post: { appAction: AppAction; itemId: string };
    };
  }>();
```

as well as lines 31-36, I think no conversion is involved so the types are still entity-level:
```ts
const appAction = await appActionRepository.post(itemId, actorId, body);
    await this.hooks.runPostHooks('post', member, repositories, {
      appAction,
      itemId,
    });
    return appAction;
```

I think my comment applies generally to all services w.r.t to their underlying entities.

Reply:
Yes we currently use the entity level types throughout the backend, as no mapping to service types was created in this PR.